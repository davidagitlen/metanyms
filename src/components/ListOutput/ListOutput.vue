<template>
  <div id='synonym-list' v-if='this.synonyms.length > 0'>
    <ul>
      <li 
      v-for='(synonym, index) in this.synonyms' 
      v-bind:key='index'
      @click.prevent = 'findSynonyms(synonym)'>{{synonym}}
      </li>
    </ul>
  </div>
</template>

<script>
export default {
  name: 'listoutput',
  props: ['synonyms', 'findSynonyms']
}
</script>

<style scoped>
  ul {
    margin: 25px 0px 0px 35px;
    width: max-content;
  }
  li {
    margin: 10px 0px 0px 10px;
  }
  li:hover {
    color: #3030FF;
    font-style: italic; 
    cursor: pointer;
    transition: all 0.2s ease-in-out;
  }
  #synonym-list {
    min-height: 80vh;
    max-height: max-content;
    width: 70%;
    font-size: 1.5em;
  }
</style>