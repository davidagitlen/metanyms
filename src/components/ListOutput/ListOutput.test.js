import { mount } from '@vue/test-utils';
import ListOutput from './ListOutput.vue';

describe('ListOutput', () => {

  test('renders correctly', () => {
    const wrapper = mount(ListOutput);
    expect(wrapper.element).toMatchSnapshot();
  })

})