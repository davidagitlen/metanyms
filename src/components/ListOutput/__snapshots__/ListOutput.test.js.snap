// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListOutput renders correctly 1`] = `<!---->`;
